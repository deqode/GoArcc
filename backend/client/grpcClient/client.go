package grpcClient

import (
	"alfred/config"
	"alfred/logger"
	"alfred/servers/openTracing/tracer/jaeger"
	"context"
	grpc_opentracing "github.com/grpc-ecosystem/go-grpc-middleware/tracing/opentracing"
	"go.uber.org/zap"
	"google.golang.org/grpc"
	"time"
)

//ClientContext to store context
type ClientContext struct {
	Ctx context.Context
}

//GetGrpcClientConnection is used to get the client connection
func GetGrpcClientConnection(config *config.Config) *grpc.ClientConn {
	var opts []grpc.DialOption
	opts = append(opts,
		grpc.WithUnaryInterceptor(
			//tracing for unary interceptors
			grpc_opentracing.UnaryClientInterceptor(grpc_opentracing.WithTracer(jaeger.Tracer)),
		),
		//tracing for stream interceptors
		grpc.WithStreamInterceptor(grpc_opentracing.StreamClientInterceptor(grpc_opentracing.WithTracer(jaeger.Tracer))),
	)

	//append grpc insecure
	opts = append(opts,
		grpc.WithInsecure(),
	)
	ctx, cancel := context.WithTimeout(context.Background(), time.Second*time.Duration(config.Grpc.RequestTimeout))
	defer cancel()

	// You must have some sort of OpenTracing Tracer instance on hand.
	conn, err := grpc.DialContext(ctx, config.Grpc.Host+":"+config.Grpc.Port, opts...)
	if err != nil {
		logger.Log.Fatal("did not connect", zap.Error(err))
	}
	return conn
}
