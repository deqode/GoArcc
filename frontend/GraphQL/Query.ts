import { gql } from '@apollo/client';

export const GET_REPOSITORIES = gql`
query repositories(
    $userid:String!
    $accountid:String!
    $provider:Types_Enum_GitProviders!
    ){
    repositories(
      provider:$provider,
      user_id:$userid,
      account_id:$accountid,
    ){repositories{
      name
    }}
  }
`
// todo : gql file

export const GET_BRANCHES = gql`
query repository(
    $ownerName:String!
    $repoName:String!
    $accountid:String!
    $provider:Types_Enum_GitProviders!
){
    repository(
      owner_name:$ownerName
      provider:$provider,
      repo_name:$repoName
      account_id:$accountid
    ){
      branches
    }
  }`

export const VCS_CONNECTIONS = gql`
query VCSConnections(
  $accountid:String!
  $provider:Types_Enum_GitProviders!
){
  VCSConnections(
    account_id:$accountid
    provider:$provider,
    ){
    vcs_connection{
            user_name
    }
  }
}`