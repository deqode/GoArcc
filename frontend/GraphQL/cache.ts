import { gql, InMemoryCache, makeVar } from '@apollo/client'

export const token = makeVar(null)

export default new InMemoryCache({
  typePolicies: {
    Query: {
      fields: {
        token: {
          read() {
            return token()
          },
        },
      },
    },
  },
})

export const GET_TOKEN = gql`
  query getToken {
    token @client
  }
`
