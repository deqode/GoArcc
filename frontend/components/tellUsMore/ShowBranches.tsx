import { useQuery } from '@apollo/client'
import {
  CircularProgress,
  FormControl,
  Grid,
  InputLabel,
  MenuItem,
  Select,
} from '@material-ui/core'
import React, { Dispatch, ReactElement, SetStateAction, useEffect, useState } from 'react'

import { GET_BRANCHES } from '../../GraphQL/Query'
import { useSelectStyles } from '../../styles/commonStyles'

interface ShowBranchesInput {
  ownerName: string
  repoName: string
  accountId: string
  provider: string
  setBranchName: Dispatch<SetStateAction<string>>
  setCloneUrl: Dispatch<SetStateAction<string>>
}

export default function ShowBranches({
  ownerName,
  repoName,
  accountId,
  provider,
  setBranchName,
  setCloneUrl,
}: ShowBranchesInput): ReactElement {
  const [branches, setbranches] = useState<Array<string>>([])
  const classes = useSelectStyles()

  const { loading, error, data } = useQuery(GET_BRANCHES, {
    variables: {
      ownerName,
      repoName,
      accountId,
      provider,
    },
  })

  useEffect(() => {
    // if(error)
    // TODO:pop error
    if (data && data.repository) {
      setbranches(data.repository.branches || [])
      setCloneUrl(data.repository.repo_url)
    }
  }, [data, error])

  const selectBranch = (
    e: React.ChangeEvent<{
      name?: string | undefined
      value: unknown
    }>
  ): void => {
    setBranchName(e.target.value as string)
  }

  return (
    <Grid item xs={12} md={12}>
      {loading ? (
        <CircularProgress />
      ) : (
        <FormControl variant="outlined">
          <InputLabel id="demo-simple-select-outlined-label">Branch Name</InputLabel>
          <Select
            className={classes.root}
            labelId="demo-simple-select-outlined-label"
            id="demo-simple-select-outlined"
            onChange={selectBranch}
            label="Repo Name"
            disabled={!(branches.length > 0)}>
            {branches.map((name: string) => (
              <MenuItem key={name} value={name}>
                {name}
              </MenuItem>
            ))}
          </Select>
        </FormControl>
      )}
    </Grid>
  )
}
