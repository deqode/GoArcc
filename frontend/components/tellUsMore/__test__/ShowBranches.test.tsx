import { MockedProvider } from '@apollo/client/testing'
import { act, create, ReactTestRenderer } from 'react-test-renderer'
import { GET_BRANCHES } from '../../../GraphQL/Query'
import ShowBranches from '../ShowBranches'

const mocks = [
  {
    request: {
      query: GET_BRANCHES,
      variables: {
        provider: 'provider',
        ownerName: 'ownerName',
        repoName: 'repoName,',
        accountId: 'accountId',
      },
    },
    result: {
      data: {
        repository: {
          branches: ['Branch1', 'Branch1'],
        },
        clone_url: 'Mockclone_url',
      },
    },
  },
]

const props = {
  accountId: 'accountId',
  provider: 'GITHUB',
  ownerName: 'MockownerName',
  repoName: 'MockrepoName',
  setBranchName: jest.fn(),
  setCloneUrl: jest.fn(),
}

describe('PSAButton', () => {
  let wrapper: ReactTestRenderer

  beforeEach(() => {
    act(() => {
      wrapper = create(
        <MockedProvider mocks={mocks} addTypename={false}>
          <ShowBranches {...props} />
        </MockedProvider>
      )
    })
  })

  it('renders correctly', async () => {
    expect(wrapper.toJSON()).toMatchSnapshot()
  })
})
