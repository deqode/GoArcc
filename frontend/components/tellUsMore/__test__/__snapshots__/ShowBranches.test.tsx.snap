// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PSAButton renders correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12"
>
  <div
    className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;
